whelktool/scripts/2019/05/delete-unused-encodingLevel.groovy

```diff
@@ -0,0 +1,26 @@
+/*
+ * This script sets obsolete or OCLC specfic encodingLevels to marc:AbbreviatedLevel
+ *
+ * See LXL-2546 for more info.
+ *
+ */
+
+NEW_ENCODING_LEVEL_VALUE = "marc:AbbreviatedLevel"
+PrintWriter scheduledToUpdateEncodingLevel = getReportWriter("scheduled-to-update-encodingLevel")
+
+selectBySqlWhere("""
+        collection in ('bib') AND
+        data#>>'{@graph,0,encodingLevel}' = 'marc:FullLevelInputByOclcParticipantsLocal' OR
+        data#>>'{@graph,0,encodingLevel}' = 'marc:LessThanFullLevelInputAddedFromABatchProcessLocal' OR
+        data#>>'{@graph,0,encodingLevel}' = 'marc:FullLevelInputAddedFromABatchProcessLocal' OR
+        data#>>'{@graph,0,encodingLevel}' = 'marc:LessThanFullLevelInputByOclcParticipantsLocal' OR
+        data#>>'{@graph,0,encodingLevel}' = 'marc:DeletedRecordLocal' OR
+        data#>>'{@graph,0,encodingLevel}' = 'marc:SystemIdentifiedMarcErrorInABatchloadedRecordLocal'
+    """) { data ->
```

Comment:
Do we need to change "marc:CoreLevel" as well?

Reply:
After checking with Tomas and Lotta we agreed to keep existing marc:CoreLevel but to disable the value in the viewer. At import of data with marc:CoreLevel the value will be set to marc:AbbreviatedLevel instead. All existing instances with marc:CoreLevel will revert to token 3 (marc:AbbreviatedLevel).

Reply:
Sounds good!